src/internal/clusterstate/v2.12.0/pjs.go

```diff
@@ -39,6 +39,16 @@ func createPJSSchema(ctx context.Context, env migrations.Env) error {
 			'input', 
 			'output'
 		);
+		-- PJS has a server side cache for jobs. Utilizing this cache allows PJS to skip subtrees when spawning job trees.
+		CREATE TABLE pjs.job_cache (
+			job_hash BYTEA NOT NULL PRIMARY KEY,
+			job_id BIGINT REFERENCES pjs.jobs(id) ON DELETE CASCADE,
+			cache_read BOOLEAN NOT NULL DEFAULT FALSE,
+		    cache_write BOOLEAN NOT NULL DEFAULT FALSE
```

Comment:
If proposal 3 is proposal 2 without extra RPCs: proposal 2 puts cache_write in pjs.jobs

Reply:
That's right. However, I think that's weird (and may have been overlooked). It feels more appropriate to put these all in the same table.) What do you think?

Reply:
Agreed